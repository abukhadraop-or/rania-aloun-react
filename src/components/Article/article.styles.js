import styled from 'styled-components';
import colors from 'styles/colors';

export const Container = styled.div`
  border-top: 0.13rem solid ${colors.mercury};
  padding: 1.5rem 0;
`;

export const ArticleTitle = styled.p`
  color: black;
  font-size: 1.5rem;
  font-weight: 700;
  margin: 1rem 0 0;
`;

export const ArticleLink = styled.p`
  color: ${colors.ashGrey};
  cursor: pointer;
  font-size: 0.8rem;
  margin: 0.2rem 0 0 0;
`;

export const ArticleFooter = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 1rem 0 0 0;
`;

export const ReadMoreText = styled.p`
  color: ${colors.ashGrey};
  cursor: pointer;
  font-size: 0.7rem;
  margin: 1rem 0;
`;

export const ArticleTagsContainer = styled.div`
  align-items: center;
  display: flex;
`;

export const ArticleTag = styled.p`
  border-radius: 0.8rem;
  border: 0.063rem solid ${colors.ashGrey};
  color: ${colors.ashGrey};
  cursor: pointer;
  font-size: 0.7rem;
  height: fit-content;
  margin: 0.125rem;
  padding: 0.2rem;
  width: fit-content;

  &:hover {
    border: 0.063rem solid ${colors.smokeyGrey};
  }
`;
