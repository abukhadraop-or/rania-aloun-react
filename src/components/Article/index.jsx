import React from 'react';
import PropType from 'prop-types';
import ArticleHeader from 'components/ArticleHeader';
import {
  Container,
  ReadMoreP,
  ArticleTag,
  ArticleLink,
  ArticleTitle,
  ArticleFooter,
  ArticleTagsContainer,
} from './article';

/**
 * Component holding the article card.
 *
 * @param {object} props Passed props to Article function.
 * @param {object[]} data Passed article.
 * @return {JSX.Element} Article card holding all the article's data.
 */
function Article({ data }) {
  return (
    <Container>
      <ArticleHeader article={data} />
      <ArticleTitle>{data.articleTitle}</ArticleTitle>
      <ArticleLink>{data.link}</ArticleLink>
      <ArticleFooter>
        <ReadMoreP>Read more...</ReadMoreP>
        <ArticleTagsContainer>
          {data.tags.map((tag) => (
            <ArticleTag key={tag.id}>{tag.name}</ArticleTag>
          ))}
        </ArticleTagsContainer>
      </ArticleFooter>
    </Container>
  );
}

Article.propTypes = {
  data: PropType.shape({
    articleTitle: PropType.string,
    link: PropType.string,
    tags: PropType.arrayOf(
      PropType.shape({
        id: PropType.number,
        name: PropType.string,
        map: PropType.func,
      })
    ).isRequired,
  }).isRequired,
};

export default Article;
