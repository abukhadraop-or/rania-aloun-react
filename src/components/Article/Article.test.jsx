import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Article from '.';

describe('MyComponent', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Article
        data={{
          articleTitle: 'title test',
          link: 'link test',
          tags: [{ id: 0, name: 'name test' }],
          id: 1,
          liked: 5,
          userName: 'Rania',
          publishDate: '2021-05-08T21:00:00.000Z',
        }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
