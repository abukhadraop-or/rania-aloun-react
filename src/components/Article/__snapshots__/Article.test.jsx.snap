// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent matches snapshot 1`] = `
<styled.div>
  <ArticleHeader
    id={1}
    liked={5}
    publishDate="2021-05-08T21:00:00.000Z"
    userName="Rania"
  />
  <styled.p>
    title test
  </styled.p>
  <styled.p>
    link test
  </styled.p>
  <styled.div>
    <styled.p>
      Read more...
    </styled.p>
    <styled.div>
      <styled.p
        key="0"
      >
        name test
      </styled.p>
    </styled.div>
  </styled.div>
</styled.div>
`;
